agate-rest/src/main/java/org/obiba/agate/web/rest/user/UsersPublicResource.java

```diff
@@ -130,12 +134,14 @@ public Response resetPassword(@FormParam("key") String key, @FormParam("password
   public Response create(@FormParam("username") String username, @FormParam("firstname") String firstName,
     @FormParam("lastname") String lastName, @FormParam("email") String email,
     @FormParam("application") List<String> applications, @FormParam("group") List<String> groups,
-    @FormParam("password") String password, @Context HttpServletRequest request) {
+    @FormParam("password") String password, @FormParam("reCaptchaResponse") String reCaptchaResponse, @Context HttpServletRequest request) {
     if(Strings.isNullOrEmpty(email)) throw new BadRequestException("Email cannot be empty");
 
     if(Strings.isNullOrEmpty(username)) throw new BadRequestException("User name cannot be empty");
 
-    if(new EmailValidator().isValid(username, null)) throw new BadRequestException("User name cannot be an email address.");
+    if(new EmailValidator().isValid(username, null)) throw new BadRequestException("User name cannot be an email address");
+
+    if (!Strings.isNullOrEmpty(reCaptchaResponse) && !reCaptchaService.verify(reCaptchaResponse)) throw new BadRequestException("Invalid recaptcha response");
```

Comment:
This check means that the captcha can be by-passed when providing an empty captcha response.